test_plans/exceptions.asciidoc

```diff
@@ -0,0 +1,243 @@
+:sectnums:
+:xrefstyle: short
+
+= Test plan for exceptions
+
+This is a test plan for the APIs described in SYCL 2020 section 4.13.2.
+"Exception class interface"
+
+== Testing scope
+
+=== Device coverage
+
+All of the tests described below are performed only on the default device that
+is selected on the CTS command line.
+
+
+== Tests
+
+Every method should be checked for accordance to spec:
+
+* Method existance
+* Const correctness
+* Type of result
+* Result for noexcept() operator
+
+=== sycl namespace functions
+
+==== make_error_code
+
+For every errc value check that:
+
+* Result of the call should be equal to `std::error_code(errc)`
+* `error_code::value` is equal to errc value passed
+* `error_code::category` is equal to sycl_category()
+* `error_code::default_error_condition()` is equal to
+  `std::error_condition(errc, sycl::sycl_category())`
+
+==== sycl_category()
+
+For each of the following use-cases:
+
+* using reference or copy of `error_category` retrieved at the function scope
+* using static variable in the test namespace that is constructed from sycl_category()
+  and then copied
```

Comment:
Looks like we have a reference to the abstract `std::error_category` class as result from the `sycl::sycl_category()`, so we cannot make a copy.
Instead of using static variable with a copy of value retrieved from sycl_category() I would suggest to:

- run all possible checks during construction of some object instance at global static scope, before `main()` and `TEST_NAME::run()`
- preserve results of all checks as boolean attributes of such object instance
- provide proper interpretation and failure output within `TEST_NAME::run()` using `util::logger` instance